### ISSUE TITLE
timeout error not happening in node v6

### ISSUE THREAD

**Issue by vigneshnrfs** [senecajs/seneca-amqp-transport#48]
I tried installing the latest version of seneca on node v6. 

I am using seneca-amqp-transport. When I try calling a pattern that does not even exist, the seneca instance does not even throw a timeout error. 

However, it works as expected in seneca v2.0.1.

Explicitly setting the timeout as below does not help either.

```js
var seneca = Seneca({
  tag: 'some-name',
  timeout: 500,
  log: 'silent',
  debug: {
    undead: false,
    short_logs: false,
  },
  strict: {
    result: true,
    add: false,
    find: true,
    maxloop: 11
  }
})
```

**Comment by nfantone** [comment]
Interesting. TBH, I haven't tried Seneca 3.0.0, yet - though, all unit testa run fine on it.

I'll give it a spin and get back to you.

**Comment by nfantone** [comment]
@vigneshnrfs Tried it with both node `5.12.0` and `6.5.0` + seneca `3.0.0` and all my desktop tests ran fine.

```sh
$ ~/seneca-amqp-transport on git:develop ✖︎ [15:29:12]
→ node -v
v6.5.0

$ ~/seneca-amqp-transport on git:develop ✖︎ [15:29:19]
→ npm -v
3.10.6

$ ~/seneca-amqp-transport on git:develop ✖︎ [15:29:23]
→ AMQP_URL=amqp://guest@guest:my.rabbitmq.server:5672 node examples/client.js
{"kind":"notice","notice":"hello","level":"info","seneca":"arpyjxxb899k/1472567371003/1009/3.0.0/-","when":1472567371038}


# Error stacktrace was shortened for legibility
{"notice":"seneca: Action role:create failed: [TIMEOUT]."
```



Could you try reproducing what I just did here? Just run the `examples/client.js` script without any listener. Don't forget to set `AMQP_URL` to some valid AMQP broker instance.

**Comment by vigneshnrfs** [comment]
Hi.. tried the same as above. But the result is the same. 

```
FDLMC945:amqp-test vignesh$ node -v
v6.5.0
FDLMC945:amqp-test vignesh$ npm -v
3.10.3
FDLMC945:amqp-test vignesh$ npm ls
seneca-amqp-transport@1.0.2-dev /Users/vignesh/Documents/amqp-test
├─┬ amqplib@0.4.2
│ ├── bitsyntax@0.0.4
│ ├── buffer-more-ints@0.0.2
│ ├─┬ readable-stream@1.1.14
│ │ ├── core-util-is@1.0.2
│ │ ├── inherits@2.0.1
│ │ ├── isarray@0.0.1
│ │ └── string_decoder@0.10.31
│ └── when@3.6.4
├─┬ amqpuri@1.0.2
│ ├── lodash.assign@4.2.0
│ ├── lodash.last@3.0.0
│ └── lodash.pick@4.4.0
├── bluebird@3.4.5
├─┬ chai@3.5.0
│ ├── assertion-error@1.0.2
│ ├─┬ deep-eql@0.1.3
│ │ └── type-detect@0.1.1
│ └── type-detect@1.0.0
├─┬ chai3-json-schema@1.2.1
│ ├── jsonpointer.js@0.3.0
│ └── tv4@1.0.18
├── dirty-chai@1.2.2
├─┬ eslint@3.4.0
│ ├─┬ chalk@1.1.3
│ │ ├── ansi-styles@2.2.1
│ │ ├── escape-string-regexp@1.0.5
│ │ ├── has-ansi@2.0.0
│ │ ├── strip-ansi@3.0.1
│ │ └── supports-color@2.0.0
│ ├─┬ concat-stream@1.5.1
│ │ ├─┬ readable-stream@2.0.6
│ │ │ ├── isarray@1.0.0
│ │ │ ├── process-nextick-args@1.0.7
│ │ │ └── util-deprecate@1.0.2
│ │ └── typedarray@0.0.6
│ ├─┬ debug@2.2.0
│ │ └── ms@0.7.1
│ ├─┬ doctrine@1.3.0
│ │ └── isarray@1.0.0
│ ├─┬ escope@3.6.0
│ │ ├─┬ es6-map@0.1.4
│ │ │ ├── d@0.1.1
│ │ │ ├── es5-ext@0.10.12
│ │ │ ├── es6-iterator@2.0.0
│ │ │ ├── es6-set@0.1.4
│ │ │ ├── es6-symbol@3.1.0
│ │ │ └── event-emitter@0.3.4
│ │ ├── es6-weak-map@2.0.1
│ │ └─┬ esrecurse@4.1.0
│ │   └── estraverse@4.1.1
│ ├─┬ espree@3.1.7
│ │ ├── acorn@3.3.0
│ │ └── acorn-jsx@3.0.1
│ ├── estraverse@4.2.0
│ ├── esutils@2.0.2
│ ├─┬ file-entry-cache@2.0.0
│ │ ├─┬ flat-cache@1.2.1
│ │ │ ├── circular-json@0.3.1
│ │ │ ├─┬ del@2.2.2
│ │ │ │ ├─┬ globby@5.0.0
│ │ │ │ │ ├── array-union@1.0.2
│ │ │ │ │ └── arrify@1.0.1
│ │ │ │ ├── is-path-cwd@1.0.0
│ │ │ │ ├─┬ is-path-in-cwd@1.0.0
│ │ │ │ │ └── is-path-inside@1.0.0
│ │ │ │ ├── pify@2.3.0
│ │ │ │ └─┬ pinkie-promise@2.0.1
│ │ │ │   └── pinkie@2.0.4
│ │ │ ├── graceful-fs@4.1.6
│ │ │ └── write@0.2.1
│ │ └── object-assign@4.1.0
│ ├─┬ glob@7.0.6
│ │ ├── fs.realpath@1.0.0
│ │ ├─┬ inflight@1.0.5
│ │ │ └── wrappy@1.0.2
│ │ ├─┬ minimatch@3.0.3
│ │ │ └─┬ brace-expansion@1.1.6
│ │ │   ├── balanced-match@0.4.2
│ │ │   └── concat-map@0.0.1
│ │ ├── once@1.3.3
│ │ └── path-is-absolute@1.0.0
│ ├── globals@9.9.0
│ ├── ignore@3.1.5
│ ├── imurmurhash@0.1.4
│ ├─┬ inquirer@0.12.0
│ │ ├── ansi-escapes@1.4.0
│ │ ├── ansi-regex@2.0.0
│ │ ├─┬ cli-cursor@1.0.2
│ │ │ └─┬ restore-cursor@1.0.1
│ │ │   ├── exit-hook@1.1.1
│ │ │   └── onetime@1.1.0
│ │ ├── cli-width@2.1.0
│ │ ├── figures@1.7.0
│ │ ├─┬ readline2@1.0.1
│ │ │ ├─┬ code-point-at@1.0.0
│ │ │ │ └── number-is-nan@1.0.0
│ │ │ ├── is-fullwidth-code-point@1.0.0
│ │ │ └── mute-stream@0.0.5
│ │ ├── run-async@0.1.0
│ │ ├── rx-lite@3.1.2
│ │ └── string-width@1.0.2
│ ├─┬ is-my-json-valid@2.13.1
│ │ ├── generate-function@2.0.0
│ │ ├─┬ generate-object-property@1.2.0
│ │ │ └── is-property@1.0.2
│ │ ├── jsonpointer@2.0.0
│ │ └── xtend@4.0.1
│ ├─┬ is-resolvable@1.0.0
│ │ └── tryit@1.0.2
│ ├─┬ js-yaml@3.6.1
│ │ ├─┬ argparse@1.0.7
│ │ │ └── sprintf-js@1.0.3
│ │ └── esprima@2.7.3
│ ├─┬ json-stable-stringify@1.0.1
│ │ └── jsonify@0.0.0
│ ├─┬ levn@0.3.0
│ │ ├── prelude-ls@1.1.2
│ │ └── type-check@0.3.2
│ ├─┬ mkdirp@0.5.1
│ │ └── minimist@0.0.8
│ ├── natural-compare@1.4.0
│ ├─┬ optionator@0.8.1
│ │ ├── deep-is@0.1.3
│ │ ├── fast-levenshtein@1.1.4
│ │ └── wordwrap@1.0.0
│ ├── path-is-inside@1.0.1
│ ├── pluralize@1.2.1
│ ├── progress@1.1.8
│ ├─┬ require-uncached@1.0.2
│ │ ├─┬ caller-path@0.1.0
│ │ │ └── callsites@0.2.0
│ │ └── resolve-from@1.0.1
│ ├── shelljs@0.6.1
│ ├── strip-bom@3.0.0
│ ├── strip-json-comments@1.0.4
│ ├─┬ table@3.7.8
│ │ ├── slice-ansi@0.0.4
│ │ ├── tv4@1.2.7
│ │ └── xregexp@3.1.1
│ ├── text-table@0.2.0
│ └─┬ user-home@2.0.0
│   └── os-homedir@1.0.1
├── eslint-config-seneca@3.0.0
├─┬ eslint-config-xo-space@0.14.0
│ └── eslint-config-xo@0.15.4
├─┬ eslint-plugin-hapi@4.0.0
│ ├── hapi-capitalize-modules@1.1.6
│ ├── hapi-for-you@1.0.0
│ ├── hapi-scope-start@2.1.1
│ └── no-arrowception@1.0.0
├── eslint-plugin-standard@2.0.0
├─┬ gulp@3.9.1
│ ├── archy@1.0.0
│ ├── deprecated@0.0.1
│ ├─┬ gulp-util@3.0.7
│ │ ├── array-differ@1.0.0
│ │ ├── array-uniq@1.0.3
│ │ ├── beeper@1.1.0
│ │ ├─┬ dateformat@1.0.12
│ │ │ ├── get-stdin@4.0.1
│ │ │ └─┬ meow@3.7.0
│ │ │   ├─┬ camelcase-keys@2.1.0
│ │ │   │ └── camelcase@2.1.1
│ │ │   ├─┬ loud-rejection@1.6.0
│ │ │   │ ├─┬ currently-unhandled@0.4.1
│ │ │   │ │ └── array-find-index@1.0.1
│ │ │   │ └── signal-exit@3.0.0
│ │ │   ├── map-obj@1.0.1
│ │ │   ├─┬ normalize-package-data@2.3.5
│ │ │   │ ├── hosted-git-info@2.1.5
│ │ │   │ ├─┬ is-builtin-module@1.0.0
│ │ │   │ │ └── builtin-modules@1.1.1
│ │ │   │ └─┬ validate-npm-package-license@3.0.1
│ │ │   │   ├─┬ spdx-correct@1.0.2
│ │ │   │   │ └── spdx-license-ids@1.2.2
│ │ │   │   └── spdx-expression-parse@1.0.3
│ │ │   ├─┬ redent@1.0.0
│ │ │   │ ├─┬ indent-string@2.1.0
│ │ │   │ │ └── repeating@2.0.1
│ │ │   │ └── strip-indent@1.0.1
│ │ │   └── trim-newlines@1.0.0
│ │ ├── lodash._reescape@3.0.0
│ │ ├── lodash._reevaluate@3.0.0
│ │ ├── lodash._reinterpolate@3.0.0
│ │ ├─┬ lodash.template@3.6.2
│ │ │ ├── lodash._basecopy@3.0.1
│ │ │ ├── lodash._basetostring@3.0.1
│ │ │ ├── lodash._basevalues@3.0.0
│ │ │ ├── lodash._isiterateecall@3.0.9
│ │ │ ├─┬ lodash.escape@3.2.0
│ │ │ │ └── lodash._root@3.0.1
│ │ │ ├─┬ lodash.keys@3.1.2
│ │ │ │ ├── lodash._getnative@3.9.1
│ │ │ │ ├── lodash.isarguments@3.1.0
│ │ │ │ └── lodash.isarray@3.0.4
│ │ │ ├── lodash.restparam@3.6.1
│ │ │ └── lodash.templatesettings@3.1.1
│ │ ├─┬ multipipe@0.1.2
│ │ │ └── duplexer2@0.0.2
│ │ ├── object-assign@3.0.0
│ │ ├── replace-ext@0.0.1
│ │ ├─┬ through2@2.0.1
│ │ │ └─┬ readable-stream@2.0.6
│ │ │   └── isarray@1.0.0
│ │ └─┬ vinyl@0.5.3
│ │   ├── clone@1.0.2
│ │   └── clone-stats@0.0.1
│ ├── interpret@1.0.1
│ ├─┬ liftoff@2.3.0
│ │ ├── extend@3.0.0
│ │ ├─┬ fined@1.0.1
│ │ │ ├── expand-tilde@1.2.2
│ │ │ ├── lodash.assignwith@4.2.0
│ │ │ ├── lodash.isarray@4.0.0
│ │ │ ├── lodash.isempty@4.4.0
│ │ │ └─┬ parse-filepath@1.0.1
│ │ │   ├─┬ is-absolute@0.2.5
│ │ │   │ ├─┬ is-relative@0.2.1
│ │ │   │ │ └─┬ is-unc-path@0.1.1
│ │ │   │ │   └── unc-path-regex@0.1.2
│ │ │   │ └── is-windows@0.1.1
│ │ │   ├── map-cache@0.2.2
│ │ │   └─┬ path-root@0.1.1
│ │ │     └── path-root-regex@0.1.2
│ │ ├── flagged-respawn@0.3.2
│ │ ├── lodash.isplainobject@4.0.6
│ │ ├── lodash.isstring@4.0.1
│ │ ├── lodash.mapvalues@4.6.0
│ │ └── rechoir@0.6.2
│ ├── minimist@1.2.0
│ ├─┬ orchestrator@0.3.7
│ │ ├── end-of-stream@0.1.5
│ │ ├── sequencify@0.0.7
│ │ └── stream-consume@0.1.0
│ ├── pretty-hrtime@1.0.2
│ ├── semver@4.3.6
│ ├── tildify@1.2.0
│ ├─┬ v8flags@2.0.11
│ │ └── user-home@1.1.1
│ └─┬ vinyl-fs@0.3.14
│   ├── defaults@1.0.3
│   ├─┬ glob-stream@3.1.18
│   │ ├── glob@4.5.3
│   │ ├─┬ glob2base@0.0.12
│   │ │ └── find-index@0.1.1
│   │ ├── minimatch@2.0.10
│   │ ├── ordered-read-streams@0.1.0
│   │ └── unique-stream@1.0.0
│   ├─┬ glob-watcher@0.0.6
│   │ └─┬ gaze@0.5.2
│   │   └─┬ globule@0.1.0
│   │     ├─┬ glob@3.1.21
│   │     │ ├── graceful-fs@1.2.3
│   │     │ └── inherits@1.0.2
│   │     ├── lodash@1.0.2
│   │     └─┬ minimatch@0.2.14
│   │       ├── lru-cache@2.7.3
│   │       └── sigmund@1.0.1
│   ├─┬ graceful-fs@3.0.11
│   │ └── natives@1.1.0
│   ├─┬ strip-bom@1.0.0
│   │ ├── first-chunk-stream@1.0.0
│   │ └── is-utf8@0.2.1
│   ├─┬ through2@0.6.5
│   │ └── readable-stream@1.0.34
│   └─┬ vinyl@0.4.6
│     └── clone@0.2.0
├─┬ gulp-eslint@3.0.1
│ └─┬ bufferstreams@1.1.1
│   └─┬ readable-stream@2.1.5
│     ├── buffer-shims@1.0.0
│     └── isarray@1.0.0
├─┬ gulp-if@2.0.1
│ ├── gulp-match@1.0.2
│ ├─┬ ternary-stream@2.0.0
│ │ ├─┬ duplexify@3.4.5
│ │ │ ├── end-of-stream@1.0.0
│ │ │ ├─┬ readable-stream@2.1.5
│ │ │ │ └── isarray@1.0.0
│ │ │ └── stream-shift@1.0.0
│ │ ├── fork-stream@0.0.4
│ │ ├─┬ merge-stream@1.0.0
│ │ │ └─┬ readable-stream@2.1.5
│ │ │   └── isarray@1.0.0
│ │ └─┬ through2@2.0.1
│ │   └─┬ readable-stream@2.0.6
│ │     └── isarray@1.0.0
│ └─┬ through2@2.0.1
│   └─┬ readable-stream@2.0.6
│     └── isarray@1.0.0
├─┬ gulp-istanbul@1.1.1
│ ├─┬ istanbul@0.4.5
│ │ ├── abbrev@1.0.9
│ │ ├── async@1.5.2
│ │ ├─┬ escodegen@1.8.1
│ │ │ ├── estraverse@1.9.3
│ │ │ └─┬ source-map@0.2.0
│ │ │   └── amdefine@1.0.0
│ │ ├── glob@5.0.15
│ │ ├─┬ handlebars@4.0.5
│ │ │ ├─┬ optimist@0.6.1
│ │ │ │ ├── minimist@0.0.10
│ │ │ │ └── wordwrap@0.0.3
│ │ │ ├── source-map@0.4.4
│ │ │ └─┬ uglify-js@2.7.3
│ │ │   ├── async@0.2.10
│ │ │   ├── source-map@0.5.6
│ │ │   ├── uglify-to-browserify@1.0.2
│ │ │   └─┬ yargs@3.10.0
│ │ │     ├── camelcase@1.2.1
│ │ │     ├─┬ cliui@2.1.0
│ │ │     │ ├─┬ center-align@0.1.3
│ │ │     │ │ ├─┬ align-text@0.1.4
│ │ │     │ │ │ └── longest@1.0.1
│ │ │     │ │ └── lazy-cache@1.0.4
│ │ │     │ ├── right-align@0.1.3
│ │ │     │ └── wordwrap@0.0.2
│ │ │     └── window-size@0.1.0
│ │ ├── nopt@3.0.6
│ │ └─┬ supports-color@3.1.2
│ │   └── has-flag@1.0.0
│ ├─┬ istanbul-threshold-checker@0.1.0
│ │ ├─┬ istanbul@0.3.22
│ │ │ ├─┬ escodegen@1.7.1
│ │ │ │ ├── esprima@1.2.5
│ │ │ │ ├── estraverse@1.9.3
│ │ │ │ └─┬ optionator@0.5.0
│ │ │ │   ├── fast-levenshtein@1.0.7
│ │ │ │   ├── levn@0.2.5
│ │ │ │   └── wordwrap@0.0.3
│ │ │ ├── esprima@2.5.0
│ │ │ ├─┬ fileset@0.2.1
│ │ │ │ ├── glob@5.0.15
│ │ │ │ └── minimatch@2.0.10
│ │ │ └── supports-color@3.1.2
│ │ └── lodash@3.6.0
│ ├─┬ through2@2.0.1
│ │ └─┬ readable-stream@2.0.6
│ │   └── isarray@1.0.0
│ └─┬ vinyl-sourcemaps-apply@0.2.1
│   └── source-map@0.5.6
├─┬ gulp-load-plugins@1.2.4
│ ├── array-unique@0.2.1
│ ├─┬ fancy-log@1.2.0
│ │ └── time-stamp@1.0.1
│ ├─┬ findup-sync@0.4.2
│ │ ├─┬ detect-file@0.1.0
│ │ │ └── fs-exists-sync@0.1.0
│ │ ├── is-glob@2.0.1
│ │ └─┬ resolve-dir@0.1.1
│ │   └─┬ global-modules@0.2.3
│ │     ├─┬ global-prefix@0.1.4
│ │     │ ├── ini@1.3.4
│ │     │ └── osenv@0.1.3
│ │     └── is-windows@0.2.0
│ ├─┬ gulplog@1.0.0
│ │ └── glogg@1.0.0
│ ├─┬ has-gulplog@0.1.0
│ │ └── sparkles@1.0.0
│ ├─┬ micromatch@2.3.11
│ │ ├─┬ arr-diff@2.0.0
│ │ │ └── arr-flatten@1.0.1
│ │ ├─┬ braces@1.8.5
│ │ │ ├─┬ expand-range@1.8.2
│ │ │ │ └─┬ fill-range@2.2.3
│ │ │ │   ├── is-number@2.1.0
│ │ │ │   ├─┬ isobject@2.1.0
│ │ │ │   │ └── isarray@1.0.0
│ │ │ │   ├── randomatic@1.1.5
│ │ │ │   └── repeat-string@1.5.4
│ │ │ ├── preserve@0.2.0
│ │ │ └── repeat-element@1.1.2
│ │ ├─┬ expand-brackets@0.1.5
│ │ │ └── is-posix-bracket@0.1.1
│ │ ├── extglob@0.3.2
│ │ ├── filename-regex@2.0.0
│ │ ├── is-extglob@1.0.0
│ │ ├─┬ kind-of@3.0.4
│ │ │ └── is-buffer@1.1.4
│ │ ├── normalize-path@2.0.1
│ │ ├─┬ object.omit@2.0.0
│ │ │ ├─┬ for-own@0.1.4
│ │ │ │ └── for-in@0.1.5
│ │ │ └── is-extendable@0.1.1
│ │ ├─┬ parse-glob@3.0.4
│ │ │ ├─┬ glob-base@0.3.0
│ │ │ │ └── glob-parent@2.0.0
│ │ │ └── is-dotfile@1.0.2
│ │ └─┬ regex-cache@0.4.3
│ │   ├── is-equal-shallow@0.1.3
│ │   └── is-primitive@2.0.0
│ └── resolve@1.1.7
├─┬ gulp-mocha@3.0.1
│ ├─┬ mocha@3.0.2
│ │ ├── browser-stdout@1.3.0
│ │ ├─┬ commander@2.9.0
│ │ │ └── graceful-readlink@1.0.1
│ │ ├── diff@1.4.0
│ │ ├── glob@7.0.5
│ │ ├── growl@1.9.2
│ │ ├── json3@3.3.2
│ │ ├─┬ lodash.create@3.1.1
│ │ │ ├── lodash._baseassign@3.2.0
│ │ │ └── lodash._basecreate@3.0.3
│ │ └── supports-color@3.1.2
│ ├─┬ plur@2.1.2
│ │ └── irregular-plurals@1.2.0
│ ├─┬ req-cwd@1.0.1
│ │ └─┬ req-from@1.0.1
│ │   └── resolve-from@2.0.0
│ ├─┬ temp@0.8.3
│ │ ├── os-tmpdir@1.0.1
│ │ └── rimraf@2.2.8
│ └── through@2.3.8
├─┬ gulp-release@1.1.0
│ ├─┬ fs-sync@1.0.2
│ │ ├─┬ glob@4.5.3
│ │ │ └── minimatch@2.0.10
│ │ ├── iconv-lite@0.2.11
│ │ ├── mkdirp@0.3.5
│ │ └── rimraf@2.5.4
│ ├─┬ gulp-bump@2.4.0
│ │ ├─┬ bump-regex@2.3.0
│ │ │ └── lodash.clone@4.5.0
│ │ ├─┬ plugin-error@0.1.2
│ │ │ ├─┬ ansi-cyan@0.1.1
│ │ │ │ └── ansi-wrap@0.1.0
│ │ │ ├── ansi-red@0.1.1
│ │ │ ├─┬ arr-diff@1.1.0
│ │ │ │ └── array-slice@0.2.3
│ │ │ ├── arr-union@2.1.0
│ │ │ └─┬ extend-shallow@1.1.4
│ │ │   └── kind-of@1.1.0
│ │ ├── plugin-log@0.1.0
│ │ └─┬ through2@2.0.1
│ │   └─┬ readable-stream@2.0.6
│ │     └── isarray@1.0.0
│ ├─┬ gulp-codename@0.1.1
│ │ ├─┬ detect-indent@3.0.1
│ │ │ └─┬ repeating@1.1.3
│ │ │   └── is-finite@1.0.1
│ │ ├── lodash@3.5.0
│ │ └── path@0.11.14
│ ├─┬ gulp-git@1.11.3
│ │ ├── any-shell-escape@0.1.1
│ │ └── require-dir@0.1.0
│ ├─┬ gulp-json-transform@0.4.2
│ │ ├─┬ promise@7.1.1
│ │ │ └── asap@2.0.4
│ │ └─┬ through2@2.0.1
│ │   └─┬ readable-stream@2.0.6
│ │     └── isarray@1.0.0
│ ├─┬ gulp-shell@0.5.2
│ │ └─┬ through2@2.0.1
│ │   └─┬ readable-stream@2.0.6
│ │     └── isarray@1.0.0
│ ├── run-sequence@1.2.2
│ ├── semver@5.3.0
│ └─┬ yargs@4.8.1
│   ├─┬ cliui@3.2.0
│   │ └── wrap-ansi@2.0.0
│   ├── decamelize@1.2.0
│   ├── get-caller-file@1.0.2
│   ├─┬ os-locale@1.4.0
│   │ └─┬ lcid@1.0.0
│   │   └── invert-kv@1.0.0
│   ├─┬ read-pkg-up@1.0.1
│   │ ├─┬ find-up@1.1.2
│   │ │ └── path-exists@2.1.0
│   │ └─┬ read-pkg@1.1.0
│   │   ├─┬ load-json-file@1.1.0
│   │   │ ├─┬ parse-json@2.2.0
│   │   │ │ └─┬ error-ex@1.3.0
│   │   │ │   └── is-arrayish@0.2.1
│   │   │ └── strip-bom@2.0.0
│   │   └── path-type@1.1.0
│   ├── require-directory@2.1.1
│   ├── require-main-filename@1.0.1
│   ├── set-blocking@2.0.0
│   ├── which-module@1.0.0
│   ├── window-size@0.2.0
│   ├── y18n@3.2.1
│   └─┬ yargs-parser@2.4.1
│     └── camelcase@3.0.0
├── jsonic@0.2.2
├── lodash@4.15.0
├─┬ pre-commit@1.1.3
│ ├─┬ cross-spawn@2.0.1
│ │ ├── cross-spawn-async@2.2.4
│ │ └─┬ spawn-sync@1.0.13
│ │   └── os-shim@0.1.3
│ └─┬ which@1.2.10
│   └── isexe@1.1.2
├─┬ seneca@3.0.0
│ ├─┬ eraro@0.4.1
│ │ └── lodash@2.4.2
│ ├── gate-executor@1.0.0
│ ├─┬ gex@0.2.2
│ │ └── lodash@3.10.0
│ ├─┬ lru-cache@4.0.1
│ │ ├── pseudomap@1.0.2
│ │ └── yallist@2.0.0
│ ├── nid@0.3.2
│ ├─┬ norma@0.3.0
│ │ └── lodash@2.4.2
│ ├─┬ patrun@0.5.0
│ │ └── lodash@3.10.0
│ ├── rolling-stats@0.1.1
│ ├─┬ seneca-transport@2.1.0
│ │ ├─┬ ndjson@1.4.3
│ │ │ └── split2@0.2.1
│ │ ├── qs@6.2.1
│ │ ├─┬ reconnect-core@1.3.0
│ │ │ └─┬ backoff@2.5.0
│ │ │   └── precond@0.2.3
│ │ └─┬ wreck@9.0.0
│ │   ├── boom@3.2.2
│ │   └── hoek@4.0.2
│ └─┬ use-plugin@0.3.1
│   └── lodash@2.4.2
├── shortid@2.2.6
├─┬ sinon@1.17.5
│ ├── formatio@1.1.1
│ ├── lolex@1.3.2
│ ├── samsam@1.1.2
│ └── util@0.10.3
├── sinon-bluebird@3.0.2
└── sinon-chai@2.8.0

FDLMC945:amqp-test vignesh$ AMQP_URL=amqp://guest:guest@localhost:5672 examples/client.js 
{"kind":"notice","notice":"hello","level":"info","seneca":"l33f7ne3ckam/1472627245919/37716/3.0.0/-","when":1472627245971}
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
^C
FDLMC945:amqp-test vignesh$ 

```

The 'Run...' message is emitted everytime setInterval function executes.

**Comment by nfantone** [comment]
@vigneshnrfs I believe I know what's going on. On Seneca `3.0.0` the default timeout has doubled. 
Take a look at (this commit)[https://github.com/senecajs/seneca/commit/03e97cdb36ac47d5182db7f8a2893294945c98a3]

So, could you try waiting longer on your first run? It should take half the time for a TIMEOUT error to arise in Seneca `2.1.0`.

**Comment by vigneshnrfs** [comment]
I've waited for about 2 minutes and no error is thrown.

I'm wondering does is it specific to amqp-transport or an issue in the seneca core module itself.

**Comment by nfantone** [comment]
> I've waited for about 2 minutes and no error is thrown.

I know this is going to sound rather insulting, but I gotta ask: are you completely positive? Because, in your example I can read 8 _"Run..."_ prints on the `3.0.1` run, which would total a waiting time of 16s - given that the interval for `setInterval` is 2000. So that _would not timeout_ in Seneca `3.0.0`, but _would_ in Seneca `2.1.0`. It explains your whole situation.

> I'm wondering does is it specific to amqp-transport or an issue in the seneca core module itself.

The transport does not manage actions specifically. And is not responsible for triggering timeouts at that level. It does, however, handle timeouts when reaching the broker or on idle connections.

**Comment by vigneshnrfs** [comment]
I am absolutely positive. I ran the scripts just before replying to you.

```
FDLMC945:amqp-test vignesh$ AMQP_URL=amqp://guest:guest@localhost:5672 examples/client.js 
{"kind":"notice","notice":"hello","level":"info","seneca":"50s7ncz4syex/1472640786756/39850/3.0.0/-","when":1472640786785}
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
Run…
^C
FDLMC945:amqp-test vignesh$ 
```

**Comment by nfantone** [comment]
Ok, I re-cloned the repo, started from scratch and now I'm getting the same behaviour you're describing. I wonder what was different on my first attempt. I'll open an issue on [senecajs/seneca](https://github.com/senecajs/seneca). Let's see what they can tell us about this.

Thanks for letting me know.

**Comment by nfantone** [comment]
@vigneshnrfs Feel free to follow this up on https://github.com/senecajs/seneca/issues/517

**Comment by vigneshnrfs** [comment]
@nfantone Thanks a lot.

**Comment by mcdonnelldean** [comment]
Folks, just a heads up, still digging here. The only changes I can see where doubling time-out and updating the Gate Executor.  Still not sure the root cause, as soon as I find it I'll let ye know.

**Comment by nfantone** [comment]
@mcdonnelldean Could you found a possible culprit on this?

@vigneshnrfs Any chance you figured this out?

**Comment by vigneshnrfs** [comment]
Haven't really looked into it. For the moment, using v2.

**Comment by nfantone** [comment]
@vigneshnrfs I'm testing with latest `3.2.1`. Seems like the issue was resolved. I'll get back to you with a report very soon.

**Comment by nfantone** [comment]
@vigneshnrfs Ok, sadly this is still going on in latest release. Already reached out to @mcdonnelldean for input.

**Comment by mcdonnelldean** [comment]
The only thing that has changed that could be responsible is the gate executor.  No other part of transport got any significant updates.  @rjrodger any thoughts on this.

Amqp is broken on 3

On 27 Sep 2016 10:15 p.m., Nicolás Fantone <notifications@github.com> wrote:

@vigneshnrfs<https://github.com/vigneshnrfs> Ok, sadly this is still going on in latest release. Already reached out to @mcdonnelldean<https://github.com/mcdonnelldean> for input.

**Comment by mcdonnelldean** [comment]
@nfantone We are going to put some time on this issue in the next couple of days, we have someone working with seneca who should be able to drill in to this and figure out just what happened.

**Comment by paolochiodi** [comment]
@nfantone @vigneshnrfs I'm able to confirm this is a bug on the seneca side that causes the timeout not to work for most actions (not only related to amqp or even transports).

The bug actually resides in seneca gate-executor, you can follow the resolution here https://github.com/senecajs/gate-executor/issues/16

**Comment by nfantone** [comment]
Replied to you in https://github.com/senecajs/gate-executor/issues/16. I'm going to leave this issue open for people coming to the transport to be aware of the bug until it's fixed.

**Comment by nfantone** [comment]
Fixed in `seneca@3.2.2` 🎉
